src/routes/settings/preferences/+page.svelte

```diff
@@ -98,44 +303,78 @@
     </SettingSection>
     <SettingSection hook="section-font" name={$_("settings.preferences.font")} description={$_("settings.preferences.fontDescription")}>
         <Select
-            hook="selector-current-font-{font.toLowerCase()}"
-            selected={font}
-            options={availableFonts}
+            hook="selector-current-font-{font.text}"
+            selected={font.value}
+            options={$availableFontsStore}
             alt
             on:change={v => {
-                UIStore.setFont(v.detail)
+                const selectedFont = $availableFontsStore.find(f => {
+                    return f.value === v.detail
+                })
+                if (selectedFont) {
+                    if (selectedFont.text === v.detail) {
+                        UIStore.setFont({ text: v.detail, value: v.detail })
+                    } else {
+                        UIStore.setFont({ text: selectedFont.text, value: v.detail })
+                    }
+                }
             }} />
-        <Button hook="button-font-open-folder" icon appearance={Appearance.Alt} tooltip={$_("generic.openFolder")}>
+        <Button
+            hook="button-font-open-folder"
+            on:click={async event => {
+                fontUpload?.click()
+            }}
+            icon
+            appearance={Appearance.Alt}
+            tooltip={$_("generic.openFolder")}>
             <Icon icon={Shape.FolderOpen} />
         </Button>
+        <input data-cy="input=upload-files" style="display:none" multiple type="file" on:change={e => saveFile(e)} bind:this={fontUpload} />
     </SettingSection>
     <SettingSection hook="section-emoji-font" name={$_("settings.preferences.emojiFont")} description={$_("settings.preferences.emojiFontDescription")}>
         <span data-cy="emoji-font-random-emoji" class="emoji">{randomEmoji}</span>
         <Select
             hook="selector-current-emoji-font-{emojiFont.toLowerCase()}"
             selected={emojiFont}
-            options={availableEmojiFonts}
+            options={$availableEmojiStore}
             alt
             on:change={v => {
                 UIStore.setEmojiFont(v.detail)
             }} />
-        <Button hook="button-emoji-font-open-folder" icon appearance={Appearance.Alt} tooltip={$_("generic.openFolder")}>
+        <Button
+            hook="button-emoji-font-open-folder"
+            on:click={async event => {
+                // let emojiUpload = event.dataTransfer?.files
```

Comment:
do we need this?

Reply:
Thank you for the catches Sara! I'mma delete them now!